### PR TITLE
Implement /flutter for flutter.new as firebase redirect

### PR DESCRIPTION
**Pull request by parlough**
This removes the concept of a "default snippet" from the samples generation tool, instead opting for a simple `dart` and `flutter` sample that can be redirected to.

### CODE REVIEW

File: pkgs/dartpad_ui/firebase.json [hunk 1]
```diff
@@ -9,6 +9,21 @@
...
+          "source": "/dart",
+          "destination": "/?sample=dart",
+          "type": 301
+        },
+        {
+          "source": "/flutter",
+          "destination": "/?sample=flutter",
+          "type": 301
+        },
+        {
```
Comment (devoncarew) [inline pkgs/dartpad_ui/firebase.json:22]: I think this duplicates the first redirect.

File: pkgs/dartpad_ui/lib/main.dart [hunk 2]
```diff
@@ -59,10 +59,6 @@ class _DartPadAppState extends State<DartPadApp> {
         path: '/',
         builder: _homePageBuilder,
       ),
-      GoRoute(
```
Comment (devoncarew) [inline pkgs/dartpad_ui/lib/main.dart:62]: @johnpryan should likely review this - he just landed a PR that introduces a `:gistId` route: https://github.com/dart-lang/dart-pad/pull/3096 (I believe this restores functionality from before the rewrite).

File: pkgs/dartpad_ui/lib/main.dart [hunk 3]
```diff
@@ -274,14 +266,13 @@ class _DartPadMainPageState extends State<DartPadMainPage>
...
     appServices.populateVersions();
-    final fallBackSnippetType = widget.path == '/flutter' ? 'flutter' : 'dart';
     appServices
         .performInitialLoad(
             gistId: widget.gistId,
             sampleId: widget.builtinSampleId,
             flutterSampleId: widget.flutterSampleId,
             channel: widget.initialChannel,
-            fallbackSnippet: Samples.getDefault(type: fallBackSnippetType))
+            fallbackSnippet: Samples.defaultSnippet())
```
Comment (devoncarew) [inline pkgs/dartpad_ui/lib/main.dart:275]: 👍

File: pkgs/samples/tool/samples.dart [hunk 4]
```diff
@@ -153,9 +146,9 @@ ${samples.map((s) => s.toTableRow()).join('\n')}
 
   String _generateSourceContent() {
     final buf = StringBuffer('''
-// Copyright 2023 the Dart project authors. All rights reserved.
-// Use of this source code is governed by a BSD-style license
-// that can be found in the LICENSE file.
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
```
Comment (devoncarew) [inline pkgs/samples/tool/samples.dart:149]: Thanks for normalizing this!

File: pkgs/samples/lib/default_dart.dart [hunk 5]
```diff
@@ -1,5 +1,9 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
```
Comment (devoncarew) [inline pkgs/samples/lib/default_dart.dart:1]: One drawback of having the copyright header here (and for default_flutter.dart) is that they'll show up for most dartpad snippets. I think it's ok to omit here, but to keep it for the more specific samples (sunflower, ...).

**Review by devoncarew — APPROVED** [review]
lgtm, but we should confirm w/ John how this'll interact with the new /:gistId route.

File: pkgs/samples/lib/default_dart.dart (hunk 5 above)
Comment (parlough) [reply pkgs/samples/lib/default_dart.dart:1]: In this PR, I ended up stripping the copyright from each snippet when loaded into DartPad, but keeping it in the source. Is there any concern with that? As the license comments end up distracting from the samples anyway.

**Review by johnpryan — APPROVED** [review]
LGTM, thanks for cleaning this up!

File: pkgs/samples/lib/default_dart.dart (hunk 5 above)
Comment (devoncarew) [reply pkgs/samples/lib/default_dart.dart:1]: Ah, I'd missed that! No real concerns.

**Pull request merged**